Tests/test_lib_pack.py

```diff
@@ -1,820 +1,835 @@
+        ("I;16L", "I;16L", 2, [0x0201, 0x0403, 0x0605]),
+        ("I;16", "I;12", 2, [0x0010, 0x0203, 0x0040]),
+    ]
+    if is_big_endian():
+        params += [
+            ("I;16", "I;16N", 2, [0x0102, 0x0304, 0x0506]),
+            ("I;16B", "I;16N", 2, [0x0102, 0x0304, 0x0506]),
+            ("I;16L", "I;16N", 2, [0x0102, 0x0304, 0x0506]),
+            ("I;16N", "I;16N", 2, [0x0102, 0x0304, 0x0506]),
+        ]
+    else:
+        params += [
+            ("I;16", "I;16N", 2, [0x0201, 0x0403, 0x0605]),
+            ("I;16B", "I;16N", 2, [0x0201, 0x0403, 0x0605]),
+            ("I;16L", "I;16N", 2, [0x0201, 0x0403, 0x0605]),
+            ("I;16N", "I;16N", 2, [0x0201, 0x0403, 0x0605]),
+        ]
+
+    # Mode CMYK
+    params += [
+        ("CMYK", "CMYK;16L", 8, [(2, 4, 6, 8), (10, 12, 14, 16)]),
+        ("CMYK", "CMYK;16B", 8, [(1, 3, 5, 7), (9, 11, 13, 15)]),
+    ]
+    if is_big_endian():
+        params += [
+            ("CMYK", "CMYK;16N", 8, [(1, 3, 5, 7), (9, 11, 13, 15)]),
+        ]
+    else:
+        params += [
+            ("CMYK", "CMYK;16N", 8, [(2, 4, 6, 8), (10, 12, 14, 16)]),
+        ]
+
+    return params
+
+
+@pytest.mark.parametrize("mode, rawmode, data, pixels", get_unpack_parameters())
+def test_unpack(
+    mode: str, rawmode: str, data: int | bytes, pixels: list[float | tuple[int, ...]]
+) -> None:
+    """
+    Test unpacking from {rawmode} to {mode}.
+    data - Either the number of bytes in a pixel of the rawmode, or a byte string.
+           If a number is given, bytes will be generated starting at 1.
+           The byte string will be used to populate the starting image.
+    pixels - The pixels to be compared against the image pixels after unpacking.
+    """
+    if isinstance(data, int):
+        data_len = data * len(pixels)
+        data = bytes(range(1, data_len + 1))
+
+    im = Image.frombytes(mode, (len(pixels), 1), data, "raw", rawmode, 0, 1)
+
+    for x, pixel in enumerate(pixels):
+        assert pixel == im.getpixel((x, 0))
+
+
+@pytest.mark.parametrize(
+    "mode, rawmode, data, pixels",
+    (
+        # Mode BGR
+        ("BGR;15", "BGR;15", 3, [(8, 131, 0), (24, 0, 8), (41, 131, 8)]),
+        ("BGR;16", "BGR;16", 3, [(8, 64, 0), (24, 129, 0), (41, 194, 0)]),
+        ("BGR;24", "BGR;24", 3, [(1, 2, 3), (4, 5, 6), (7, 8, 9)]),
+    ),
+)
+def test_unpack_deprecated(
+    mode: str, rawmode: str, data: int | bytes, pixels: list[float | tuple[int, ...]]
+) -> None:
+    with pytest.warns(DeprecationWarning):
+        test_unpack(mode, rawmode, data, pixels)
+
+
+@pytest.mark.parametrize(
+    "mode, rawmode, data, pixels",
+    (
+        ("L", "L", 0, [0]),
+        ("RGB", "RGB", 2, [0]),
+        ("CMYK", "CMYK", 2, [0]),
+    ),
+)
+def test_unpack_valueerror(
+    mode: str, rawmode: str, data: int | bytes, pixels: list[float | tuple[int, ...]]
+) -> None:
+    with pytest.raises(ValueError):
+        test_unpack(mode, rawmode, data, pixels)
```

Comment:
Calling one test from another seems odd, and a potential source of confusion?

Reply:
I could make `test_unpack` not a test function, and have all three test functions call that, but I don't see how that would be any clearer.